src/operator/nn/batch_norm.cc

```diff
@@ -452,19 +453,66 @@ static inline bool BatchNormStorageType(const nnvm::NodeAttrs &attrs,
                                         std::vector<int> *out_attrs) {
   CHECK_EQ(in_attrs->size(), 5);
   CHECK_EQ(out_attrs->size(), 3);
-  return MKLDNNStorageType(attrs, dev_mask, true, dispatch_mode,
-                           in_attrs, out_attrs);
+  const BatchNormParam &param = nnvm::get<BatchNormParam>(attrs.parsed);
+
+  if ((common::ContainsStorageType(*in_attrs, kRowSparseStorage) ||
+       common::ContainsStorageType(*in_attrs, kCSRStorage)) &&
+      param.fix_gamma) {
+    LOG(FATAL) << "fix_gamma=True is not supported for sparse ndarrays. Tracked at #11647";
+  }
+  for (int& v : *in_attrs)
+    if (v == - 1) v = kDefaultStorage;
+  bool dispatched = false;
+  if (!dispatched && (common::ContainsStorageType(*in_attrs, kRowSparseStorage) ||
+                      common::ContainsStorageType(*in_attrs, kCSRStorage))) {
+    dispatched = dispatch_fallback(out_attrs, dispatch_mode);
+  }
+#if MXNET_USE_MKLDNN == 1
```

Comment:
do we need this if else here. MKLDNNStorageType is handling both cases.

Reply:
MKLDNNStorageType is not defined if MXNET_USE_MKLDNN=0

Reply:
okay.